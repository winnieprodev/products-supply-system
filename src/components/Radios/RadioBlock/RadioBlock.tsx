import { useCallback, useContext } from "react";

import { RadiosContext } from "../RadiosContext";
import * as Styled from "./radioBlock.styled";

interface OwnProps {
  index: string;
}

const RadioBlock: React.FC<React.PropsWithChildren<OwnProps>> = ({
  index,
  children,
}) => {
  const { selected, setSelected } = useContext(RadiosContext);
  const onClick = useCallback(() => setSelected(index), [setSelected, index]);
  return (
    <Styled.Block $selected={selected === index} onClick={onClick}>
      <Styled.Indicator $selected={selected === index} />
      <Styled.Content>{children}</Styled.Content>
    </Styled.Block>
  );
};

export default RadioBlock;
