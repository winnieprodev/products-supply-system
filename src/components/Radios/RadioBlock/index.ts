export { default as RadioBlock } from "./RadioBlock";
