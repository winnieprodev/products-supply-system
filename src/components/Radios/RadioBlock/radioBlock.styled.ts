import styled, { css } from "styled-components";
import {
  primaryColor,
  secondaryColor,
  secondaryColorDarker,
  secondaryColorLighter,
} from "styles/colors";

interface BlockProps {
  $selected: boolean;
}

export const Block = styled.div<BlockProps>`
  display: flex;
  flex-grow: 1;
  width: 100px;
  max-width: 33.3%;
  padding: 12px;
  align-items: center;
  cursor: pointer;
  background-color: ${({ $selected }) =>
    $selected ? secondaryColor : "#fdfdfd"};
  border: 1px solid ${secondaryColor};
  border-radius: 10px;
  transition: background ease 0.3s;

  &:hover {
    background-color: ${secondaryColorLighter};
  }
  &:active {
    background-color: ${secondaryColorDarker};
  }
`;

interface IndicatorProps {
  $selected: boolean;
}
export const Indicator = styled.div<IndicatorProps>`
  border: 1px solid #b8b8b8;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin-right: 8px;

  ${({ $selected }) =>
    $selected &&
    css`
      position: relative;

      &:before {
        position: absolute;
        display: block;
        content: "";
        border-radius: 50%;
        background: ${primaryColor};
        left: 2px;
        top: 2px;
        bottom: 2px;
        right: 2px;
      }
    `}
}
`;

export const Content = styled.div`
  width: calc(100% - 16px);
`;
