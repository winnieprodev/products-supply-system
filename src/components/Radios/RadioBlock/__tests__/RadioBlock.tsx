import { render } from "@testing-library/react";

import RadioBlock from "../RadioBlock";

describe("RadioBlock", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(<RadioBlock index="a">test</RadioBlock>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
