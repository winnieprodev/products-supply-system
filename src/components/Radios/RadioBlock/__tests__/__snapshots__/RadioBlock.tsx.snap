// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioBlock Snapshot tests it matches the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  width: 100px;
  max-width: 33.3%;
  padding: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  background-color: #fdfdfd;
  border: 1px solid #EEEEEE;
  border-radius: 10px;
  -webkit-transition: background ease 0.3s;
  transition: background ease 0.3s;
}

.c0:hover {
  background-color: #F3F3F3;
}

.c0:active {
  background-color: #E7E7E7;
}

.c1 {
  border: 1px solid #b8b8b8;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.c2 {
  width: calc(100% - 16px);
}

<div
  class="c0"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    test
  </div>
</div>
`;
